crates/girlmath/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+///
+/// # Arguments
+/// * `amt` - A string slice that holds the amount in major currency units.
+/// * `currency_code` - The ISO or cryptocurrency code as a string slice.
+///
+/// # Returns
+/// A `Result<String, String>` which is Ok with the amount in minor units
+/// or an Err with an error message.
+///
+/// # Examples
+/// ```
+/// use girlmath::major_to_minor;
+///
+/// let cents = major_to_minor("10.50", "USD").unwrap();
+/// assert_eq!("1050", cents);
+/// ```
+pub fn major_to_minor(amt: &str, currency_code: &str) -> Result<String, String> {
+    let sig_digs = iso::find(currency_code)
+        .map(|c| c.exponent)
+        .or_else(|| crypto::find(currency_code).map(|c| c.exponent))
+        .ok_or_else(|| "unsupported currency code".to_string())?;
+
+    let multiplier = Decimal::from(10u32.pow(sig_digs));
+    let amt_dec = Decimal::from_str(amt).map_err(|_| "invalid amount".to_string())?;
+
+    let minor_amt = (multiplier * amt_dec).trunc();
+
+    Ok(minor_amt.to_string())
+}
+
+/// Converts a minor currency unit (like cents) to a major currency unit (like dollars).
+///
+/// # Arguments
+/// * `amt` - A string slice that holds the amount in minor currency units.
+/// * `currency_code` - The ISO or cryptocurrency code as a string slice.
+///
+/// # Returns
+/// A `Result<String, String>` which is Ok with the amount in major units
+/// or an Err with an error message.
+///
+/// # Examples
+/// ```
+/// use girlmath::minor_to_major;
+///
+/// let dollars = minor_to_major("1050", "USD").unwrap();
+/// assert_eq!("10.50", dollars);
+/// ```
+pub fn minor_to_major(amt: &str, currency_code: &str) -> Result<String, String> {
+    let sig_digs = iso::find(currency_code)
+        .map(|c| c.exponent)
+        .or_else(|| crypto::find(currency_code).map(|c| c.exponent))
+        .ok_or_else(|| "unsupported currency code".to_string())?;
+
+    let multiplier = Decimal::from(10u32.pow(sig_digs));
+    let amt_dec = Decimal::from_str(amt).map_err(|_| "invalid amount".to_string())?;
+
+    let major_amt = (amt_dec / multiplier).trunc_with_scale(sig_digs);
+
+    Ok(major_amt.to_string())
+}
+
+/// Converts an amount from one currency to another using a given exchange rate.
+///
+/// # Arguments
+/// * `payin_amt` - The amount in the pay-in currency as a string slice.
+/// * `payin_currency` - The pay-in currency code as a string slice (currently unused).
+/// * `payout_currency` - The pay-out currency code as a string slice.
+/// * `rate` - The exchange rate as a string slice.
+///
+/// # Returns
+/// A `Result<String, String>` which is Ok with the converted amount
+/// or an Err with an error message.
+///
+/// # Examples
+/// ```
+/// use girlmath::convert;
+///
+/// let payout_amt = convert("3", "USD", "GHS", "12.10").unwrap();
+/// assert_eq!("36.30", payout_amt);
+/// ```
+pub fn convert(
+    payin_amt: &str,
+    _payin_currency: &str,
+    payout_currency: &str,
+    rate: &str,
+) -> Result<String, String> {
```

Comment:
Maybe `payin_currency` isn't needed at all? I'm not seeing it used here 🤔

Reply:
https://github.com/TBD54566975/girlmath/pull/6#discussion_r1423460747